Assets/MixedRealityToolkit.SDK/Experimental/Dialog/Scripts/Dialog.cs

```diff
@@ -0,0 +1,232 @@
+        {
+            return;
+        }
+
+        /// <summary>
+        /// Generates buttons - Must parent them under buttonParent!
+        /// </summary>
+        protected abstract void GenerateButtons();
+
+        /// <summary>
+        /// Lays out the buttons on the dialog
+        /// Eg using an ObjectCollection
+        /// </summary>
+        protected abstract void FinalizeLayout();
+
+        /// <summary>
+        /// Set the title and message using the result
+        /// Eg using TextMesh components 
+        /// </summary>
+        protected abstract void SetTitleAndMessage();
+
+        /// <summary>
+        /// Instantiates a dialog and passes it a result
+        /// </summary>
+        /// <param name="dialogPrefab"></param>
+        /// <param name="result"></param>
+        /// <returns></returns>
+        public static Dialog Open(GameObject dialogPrefab, DialogResult result)
+        {
+            GameObject dialogGo = GameObject.Instantiate(dialogPrefab) as GameObject;
+            Dialog dialog = dialogGo.GetComponent<Dialog>();
+
+            dialog.Launch(result);
+            return dialog;
+        }
+
+        /// <summary>
+        /// Instantiates a dialog and passes a generated result
+        /// </summary>
+        /// <param name="dialogPrefab"></param>
+        /// <param name="buttons"></param>
+        /// <param name="title"></param>
+        /// <param name="message"></param>
+        /// <returns></returns>
+        public static Dialog Open(GameObject dialogPrefab, DialogButtonType buttons, string title, string message)
+        {
+            GameObject dialogGameObject = GameObject.Instantiate(dialogPrefab) as GameObject;
+            Dialog dialog = dialogGameObject.GetComponent<Dialog>();
+
+            DialogResult result = new DialogResult
+            {
+                Buttons = buttons,
+                Title = title,
+                Message = message
+            };
+
+            dialog.Launch(result);
+            return dialog;
+        }
+
+        public static Dialog Open(GameObject dialogPrefab, DialogButtonType buttons, string title, string message, System.Object variable)
+        {
+            GameObject dialogGameObject = GameObject.Instantiate(dialogPrefab) as GameObject;
+            Dialog dialog = dialogGameObject.GetComponent<Dialog>();
+
+            DialogResult result = new DialogResult
+            {
+                Buttons = buttons,
+                Title = title,
+                Message = message,
+                Variable = variable
+            };
+
+            dialog.Launch(result);
+            return dialog;
+        }
+
+        // Start is called before the first frame update
+        void Start()
+        {
+
+        }
+
+        // Update is called once per frame
+        void Update()
+        {
+
+        }
+    }
+
+    /// <summary>
+    /// Enum that describes the current state of a Dialog.
+    /// </summary>
+    public enum DialogState
```

Comment:
Please move these [public enums into their own files](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html#public-classes-structs-and-enums-should-all-go-in-their-own-files).